addons/bus/static/src/js/async_systray.js

```diff
@@ -0,0 +1,146 @@
+
+const S_CREATED = "created";
+const S_PROCESSING = "processing";
+const S_DONE = "done";
+const S_FULFILLED = "fulfilled";
+const S_FAILED = "failed";
+const STATUS = [S_CREATED, S_PROCESSING, S_DONE, S_FULFILLED, S_FAILED];
+
+const statusIconMap = new Map();
+statusIconMap.set(S_CREATED, 'fa fa-arrow-circle-right');
+statusIconMap.set(S_PROCESSING, 'fa fa-spinner fa-spin');
+statusIconMap.set(S_DONE, 'fa fa-check-circle');
+statusIconMap.set(S_FULFILLED, 'fa fa-check-circle text-success');
+statusIconMap.set(S_FAILED, 'fa fa-exclamation-circle text-danger');
+
+const statusTooltipMap = new Map();
+statusTooltipMap.set(S_CREATED, _lt('The task has been scheduled for processing and will start shortly.'));
+statusTooltipMap.set(S_PROCESSING, _lt('The task is processing.'));
+statusTooltipMap.set(S_DONE, _lt('The task has been processed, click to resume the execution.'));
+statusTooltipMap.set(S_FULFILLED, _lt('The task has be completely executed, re-click to re-execute it.'));
+statusTooltipMap.set(S_FAILED, _lt('The task failed due to an error, click to reveal it.'));
+
+const notifHead = _lt("Background task completed");
+const notifDoneBody = _lt("The task %s is ready, you can resume its execution via the task list.");
+const notifErrorBody = _lt("The task %s failed, you can show the error via the task list.");
+const notifFulfillBody = _lt("The task %s has been completed by the server.")
+
+
+const AsyncSpinner = Widget.extend({
+    name: 'async_spinner',
+    template: 'bus.systray.AsyncSpinner',
+    events: {
+        'click .o_async_job_status_done': '_onClickDone',
+        'click .o_async_job_status_failed': '_onClickFailed',
+        'show.bs.dropdown': '_render',
+    },
+    init: function () {
+        this._super(...arguments);
+        this.jobs = {};
+        bus.on('async_job_created', this, this._updateJob.bind(this, S_CREATED));
+        bus.on('async_job_processing', this, this._updateJob.bind(this, S_PROCESSING));
+        bus.on('async_job_done', this, this._updateJob.bind(this, S_DONE));
+        bus.on('async_job_fulfilled', this, this._updateJob.bind(this, S_FULFILLED));
+        bus.on('async_job_failed', this, this._updateJob.bind(this, S_FAILED));
+    },
+    start: function () {
+        this._selectSpinner('.o_async_finish');
+        return this._super(...arguments);
+    },
+    _onClickDone(ev) {
+        const $target = $(ev.currentTarget).closest(
+            '.o_async_job_status_done, .o_async_job_status_fulfilled');
+        const job = this.jobs[$target.data("jobid")];
+        this.do_action(job.payload.result).then(_ => {
+            this._rpc({
+                model: 'ir.async',
+                method: 'fulfill',
+                args: [job.id, job.name],
+            });
+        }, _ => {
+            this._updateJob(S_FAILED, job.id, job.name, null);
+        });
+    },
+    _onClickFailed(ev) {
+        const $target = $(ev.currentTarget).closest('.o_async_job_status_failed');
```

Comment:
looks useless, since its currentTarget who has `data-job-id`

Reply:
I have the ev.currentTarget on one of the child element even if the handler is set on the parent. I used closest in order to ensure I'm on the parent that has the data-job-id :)